### PR TITLE
virtual/libudev & revdeps: drop eudev support from the newest version, account for disabled static-libs

### PR DESCRIPTION
**Pull request by mgorny**
@floppym, even if it turns out we can reenable static-libs on udev/systemd, I think providers should assume `(-)` for that flag.

### CODE REVIEW

**Comment by floppym** [comment]
> @floppym, even if it turns out we can reenable static-libs on udev/systemd, I think providers should assume (-) for that flag.

Yes, that makes sense.

**Comment by gentoo-bot2** [comment]
## Pull Request assignment

*Submitter*: @mgorny
*Areas affected*: ebuilds
*Packages affected*: dev-libs/libgudev, dev-libs/libusb, sys-apps/pciutils, sys-fs/cryptsetup, sys-fs/lvm2...

**dev-libs/libgudev**: @floppym, @gentoo/gnome
**dev-libs/libusb**: @gentoo/base-system, ~~peter[at]stuge.se~~
**sys-apps/pciutils**: @gentoo/base-system
**sys-fs/cryptsetup**: @gentoo/base-system
**sys-fs/lvm2**: @robbat2, ~~agk[at]redhat.com~~
**sys-fs/zfs**: @ryao, @gyakovlev
**virtual/libudev**: @gentoo/udev, @gentoo/eudev, @gentoo/systemd

## Linked bugs
Bugs linked: [697550](https://bugs.gentoo.org/697550)

---
In order to force reassignment and/or bug reference scan, please append `[please reassign]` to the pull request title.

*Docs*: [Code of Conduct](https://wiki.gentoo.org/wiki/Project:Council/Code_of_conduct) ● [Copyright policy](https://www.gentoo.org/glep/glep-0076.html) ([expl.](https://dev.gentoo.org/~mgorny/articles/new-gentoo-copyright-policy-explained.html)) ● [Devmanual](https://devmanual.gentoo.org/) ● [GitHub PRs](https://wiki.gentoo.org/wiki/Project:GitHub/Pull_requests) ● [Proxy-maint guide](https://wiki.gentoo.org/wiki/Project:Proxy_Maintainers/User_Guide)

File: virtual/libudev/libudev-232-r1.ebuild [hunk 1]
```diff
@@ -1,24 +1,15 @@
...
-		static-libs? (
-			>=sys-fs/eudev-1.3:0/0[${MULTILIB_USEDEP},static-libs(-)]
-		)
-		!static-libs? ( || (
-			>=sys-fs/eudev-1.3:0/0[${MULTILIB_USEDEP}]
-			>=sys-fs/udev-232:0/0[${MULTILIB_USEDEP}]
-		) )
-	)
+	!systemd? ( >=sys-fs/udev-232:0/0[${MULTILIB_USEDEP}] )
 	systemd? ( >=sys-apps/systemd-212-r5:0/2[${MULTILIB_USEDEP}] )
```
Comment (floppym) [inline virtual/libudev/libudev-232-r1.ebuild]: This should probably be `>=sys-apps/systemd-232` instead.

Comment (mgorny) [reply virtual/libudev/libudev-232-r1.ebuild]: Fixed.

**Comment by floppym** [comment]
static-libs support has been restored in systemd-243-r2 and udev-243-r1. We should revbump or version bump virtual/libudev with `~arch` keywords for that.

**Comment by mgorny** [comment]
I presume that 'we' is politically-neutral pronoun for 'you' here ;-).

**Comment by blueness** [comment]
I should have a new release of eudev that will provide version 232.

**Comment by blueness** [comment]
This should be fixed by eudev-3.2.9 which I just added to the tree.  As far as I can see it current with systemd's version 243 which is well above 232.  Can we hold off on this until eudev-3.2.9 is stabilized please.

**Comment by mattst88** [comment]
I don't have any skin in the game, but at this point it's been six weeks since https://bugs.gentoo.org/694014 was filed and you just finally responded to it when patches appeared to drop eudev from virtual/libudev. Why should eudev remain in virtual/libudev when you aren't responsive to bug reports?

Similarly, https://bugs.gentoo.org/666892 comes to mind as another example.

**Comment by blueness** [comment]
@mattst88 Currently eudev is the only provider that builds on musl and uclibc systems.  I'll recruit help to make sure we don't fall behind again.

**Comment by mgorny** [comment]
eudev & musl aren't stable. I'll readd eudev to the fresh ~arch release.

I don't think we should keep unmaintained udev fork in stable, and especially advertise it as the default choice.

**Comment by blueness** [comment]
@mgorny I was never for advertising eudev as the default.  This happened was after some long debate on gentoo-dev@ and I forget who wanted it or why.  However, not stabilizing eudev and keeping it ~arch will break all my uclibc and musl catalyst builds.  Also, I think we need to tell everyone on gentoo-dev@, especially releng that we're switching back to sys-fs/udev as the default provider.  It will have an impact on everyone.

I know the rest of the gentoo community may not be interested in embedded, but right now gentoo/musl is in commercial use.  I've tried to be a unintrusive as possible, but again, I need eudev stabilized.

**Comment by hydrapolic** [comment]
> eudev & musl aren't stable. I'll readd eudev to the fresh ~arch release.

sys-fs/eudev-3.2.5 *is* stable. If you remove eudev from stable, please look into https://bugs.gentoo.org/681586 to not break our systems.

**Pull request closed without merging**